### PR TITLE
feat(orch): packet affordances for `LocalOrchestrationAccount`

### LINKED ISSUE #9783: orchestration transfer() vow should not resolve until the remote chain confirms
## What is the Problem Being Solved?

`transfer` method in `local-orchestration-account.js` returns a vow that returns once the message is put in the outbox. The transfer isn't really complete until the remote chain confirms. Moreover the transfer may fail and in that case the vow should reject.

Same goes for `cosmos-orchestration-account.js` but that's "Not yet implemented". I wrote the PR title generically as a product requirement. 

## Description of the Design

Prior art: https://gist.github.com/michaelfig/280acc7d0bb1eb10bf7570929305bb80

Monitor transfers accepts a single “app” / “tap” at a time. If we use it in a platform-level service that watches outgoing transfers and resolves promises, we still need to support developers registering their own handlers. We could use the one slot with a generic handler that calls out to other handlers.

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #9783

## Description

Introduce a per-LOA `packetTools` to grant powers used by IBC-specific `makeIBCReplyKit` and `makeIBCTransferSender` which enables a `localOrchestrationAccount` to send a packet and wait for a corresponding reply packet (such as IBC acknowledgement or IBC timeout) in the context of a single localChainAccount address.

The most important files are `agoric-sdk/packages/orchestration/src/exos/packet-tools.js` and `ibc-packet.js` in the same directory.

### Security Considerations

This change is the next layer wrapping localchain and vtransfer powers, but in an attenuated fashion.  They don't introduce more powers than are already available via `LOA.monitorTransfers(...)`.

### Scaling Considerations

More message exchanges, but most of those are caused by inbound Cosmos transactions, so they should not consume much CPU per block.

### Documentation Considerations

New API surface.  Existing data and deployments will not be affected, especially since the orchestration vat+API has not yet landed on chain.

### Testing Considerations

Unit and bootstrap tests have been implemented.  The `agoric-sdk/packages/boot/test/supports.ts` fake bridge has been reorganised to greater mimic the actual chain and make tests pass.

### Upgrade Considerations

Some more Exos shipped with the Orch API, with the usual considerations of adding new Exo singletons and classKits and maintaining them.

### CODE REVIEW

**Comment by turadg** [comment]
[7af594c](https://github.com/Agoric/agoric-sdk/pull/9820/commits/7af594c0870eba089815511f438aac08bb0f42b3) failed with,
```
RemoteError(error:liveSlots:v20#70003)#10 ERROR_NOTE: Sent as error:liveSlots:v1#70001
Logging sent error stack (RemoteError(error:liveSlots:v20#70003)#10)
not ok 3 - vtransfer › vtransfer %ava-dur=41765ms
#   REJECTED from ava test("vtransfer"): (RemoteError(error:liveSlots:v1#70001)#11)
```
but passed for me locally. I'm [rerunning](https://github.com/Agoric/agoric-sdk/actions/runs/10199787318/job/28220725187?pr=9820) it now. If it succeeds then we have a new flake.

**Review by dckc — COMMENTED** [review]
I don't understand the premise of this PR. `transfer()` uses `executeTx` which says it waits for an ack. Why do we think it doesn't?

https://github.com/Agoric/agoric-sdk/issues/9783#issuecomment-2269440608

**Comment by michaelfig** [comment]
> I don't understand the premise of this PR. `transfer()` uses `executeTx` which says it waits for an ack. Why do we think it doesn't?

`executeTx` responses are "prompt" (i.e. have no dependence on I/O, just the scheduler) upon receipt of the transaction in the local chain.  IBC "acks" are multichain: they don't happen until an outbound IBC packet is relayed to its destination chain, and then the destination chain sends a reply packet in some future block to be relayed back to the initiating chain.

**Comment by dckc** [comment]
How about fixing the local-chain `executeTx` docs in this PR then?

IOU a separate issue for: maybe we shouldn't use the same `executeTx` name for the local and interchain account cases, since what they do is so different.

**Comment by dckc** [comment]
suggestion:

```diff
@@ -151,10 +151,13 @@ export const prepareLocalChainAccountKit = (zone, { watch }) =>
           return E(purse).withdraw(amount);
         },
         /**
-         * Execute a batch of messages in order as a single atomic transaction
-         * and return the responses. If any of the messages fails, the entire
-         * batch will be rolled back. Use `typedJson()` on the arguments to get
-         * typed return values.
+         * Execute a batch of messages on the local chain. Note in particular,
+         * that for an IBC transfer, executing a transaction only queues a
+         * transfer.
+         *
+         * Messages are executed in order as a single atomic transaction and
+         * return the responses. If any of the messages fails, the entire batch
+         * will be rolled back on the local chain.
          *
          * @template {TypedJson[]} MT messages tuple (use const with multiple
          *   elements or it will be a mixed array)
@@ -162,6 +165,7 @@ export const prepareLocalChainAccountKit = (zone, { watch }) =>
          * @returns {PromiseVowOfTupleMappedToGenerics<{
          *   [K in keyof MT]: JsonSafe<ResponseTo<MT[K]>>;
          * }>}
+         * @see {typedJson} which can be used on arguments to get typed return values.
          */
         async executeTx(messages) {
           const { address, system } = this.state;
```

File: packages/boot/test/orchestration/restart-contracts.test.ts [hunk 1]
```diff
@@ -18,8 +18,9 @@ test.before(async t => {
 });
 test.after.always(t => t.context.shutdown?.());
 
-// Not interesting because it doesn't wait on other chains. Leaving here because maybe it will before it's done.
-test.serial('sendAnywhere', async t => {
+// FIXME the test needs to be able to send the acknowledgementPacket ack
+// so that the transfer vow resolves.
+test.serial.failing('sendAnywhere', async t => {
```
Comment (dckc) [inline packages/boot/test/orchestration/restart-contracts.test.ts:23]: I'm a little hesitant to leave this failing.

**Review by dckc — APPROVED** [review]
I'm not confident about my ability to verify the correctness of all the watcher code patterns (feels like a weird form of CPS) but `test('transfer', ...)` pretty clearly tests this feature.

**Pull request merged**